cg/apps/demultiplex/sample_sheet/sample_sheet_creator.py

```diff
@@ -112,17 +128,17 @@ def process_samples_for_sample_sheet(self) -> None:
         else:
             LOG.info("Skipped adding dummy samples since they are not needed")
         self.remove_unwanted_samples()
-        samples_in_lane: List[FlowCellSample]
+        samples_in_lane: List[Union[FlowCellSampleBCLConvert, FlowCellSampleBcl2Fastq]]
         reverse_complement: bool = is_reverse_complement(run_parameters=self.run_parameters)
         for lane, samples_in_lane in get_samples_by_lane(self.lims_samples).items():
-            LOG.info(
-                f"Adapting index and barcode mismatch values (if applicable) for samples in lane {lane}"
-            )
-            adapt_samples(
+            LOG.info(f"Adapting index and barcode mismatch values for samples in lane {lane}")
+            self.update_barcode_mismatch_values_for_samples(samples_in_lane)
+            update_indexes_for_samples(
                 samples=samples_in_lane,
                 run_parameters=self.run_parameters,
                 reverse_complement=reverse_complement,
             )
+        self.add_override_cycles_to_samples()
```

Comment:
here, I think that semantics is everything. We need to first add the override cycles, then we need to pad_revcomp indexes and after that we need to update the barcode mismatches. Override cycles will tell us how many index cycles the sample has. Padding should then be done afterwards, even though we currently pad based on the sample type. The last thing we want to do is to update the barcode mismatch values, once again, the way all our logic plays together now wont allow us to pad or alter the index length, but if we did, then this would also be a bug. I don't know how comprehensive the test is, so I would simply put it in the semantic order in which it could prevent bugs to begin with.

Reply:
Agree, fxed